### PR TITLE
feat: Modal.xxx supports footer

### PR DESCRIPTION
**Pull request by owjs3901**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

Modal func has not footer props. Therefore Modal footer created by modal func cannot be modified.

Finnaly I add footer props to Modal Func.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Modal.comfirm support `footer` now.   |
| 🇨🇳 Chinese |   Modal.confirm 新增 `footer` 属性以自定义按钮列表。    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
1. Please add test case.
2. Documentation should be updated.

**Comment by owjs3901** [comment]
> 1. Please add test case.
> 2. Documentation should be updated.

**Comment by owjs3901** [comment]
> 1. Please add test case.
> 2. Documentation should be updated.

OK! I will try!

File: package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "site": "dumi build && cp .surgeignore _site",
     "sort": "npx sort-package-json",
     "sort-api": "antd-tools run sort-api-table",
-    "start": "PORT=8001 dumi dev",
+    "start": "dumi dev",
```
Comment (afc163) [inline package.json:89]: 为啥改这个？

**Comment by afc163** [comment]
feature branch

File: package.json (hunk 1 above)
Comment (owjs3901) [reply package.json]: I fixed for my local pc. This commit is wrong. So I fix this.

**Comment by owjs3901** [comment]
I changed branch

File: components/config-provider/__tests__/__snapshots__/components.test.tsx.snap [hunk 2]
```diff
@@ -40596,7 +40596,7 @@ exports[`ConfigProvider components Tree configProvider componentSize large 1`] =
     <div
       aria-hidden="true"
       class="config-tree-treenode"
-      style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden;"
+      style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
```
Comment (afc163) [inline components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:40599]: 这个分支不用修改这个，rebase 一下 feature。

Comment (owjs3901) [reply components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:40599]: This PR can't pass test when do not fix snapshot.

Comment (owjs3901) [reply components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:40599]: This pr can't pass test When do not fix snapshot.

First of all, I rebase feature branch

**Comment by owjs3901** [comment]
@afc163 Completed!

File: components/modal/index.zh-CN.md [hunk 3]
```diff
@@ -100,6 +100,7 @@ demo:
 | closable | 是否显示右上角的关闭按钮 | boolean | false | 4.9.0 |
 | closeIcon | 自定义关闭图标 | ReactNode | undefined | 4.9.0 |
 | content | 内容 | ReactNode | - |  |
+| footer | 底部内容，当不需要默认底部按钮时，可以设为 `footer: null` | ReactNode | - |  |
```
Comment (afc163) [inline components/modal/index.zh-CN.md:103]:
```suggestion
| footer | 底部内容，当不需要默认底部按钮时，可以设为 `footer: null` | ReactNode | - | 5.1.0  |
```

File: components/modal/index.en-US.md [hunk 4]
```diff
@@ -97,6 +97,7 @@ The items listed above are all functions, expecting a settings object as paramet
 | closable | Whether a close (x) button is visible on top right of the confirm dialog or not | boolean | false | 4.9.0 |
 | closeIcon | Custom close icon | ReactNode | undefined | 4.9.0 |
 | content | Content | ReactNode | - |  |
+| footer | Footer content, set as `footer: null` when you don't need default buttons	| ReactNode | - |  |
```
Comment (afc163) [inline components/modal/index.en-US.md:100]:
```suggestion
| footer | Footer content, set as `footer: null` when you don't need default buttons	| ReactNode | - | 5.1.0 |
```

**Review by afc163 — APPROVED** [review]

**Pull request merged**